agent/src/beerocks/slave/beerocks_slave_main.cpp

```diff
@@ -288,12 +286,11 @@ static int run_beerocks_slave(beerocks::config_file::sConfigSlave &beerocks_slav
         }
     }
 
-    // fill configuration (only to read stop on failure attempts)
-    son::slave_thread::sSlaveConfig son_slave_conf;
-    fill_son_slave_config(beerocks_slave_conf, son_slave_conf, 0);
+    // read the number of failures allowed before stopping agent from platform configuration
+    int stop_on_failure_attempts = bpl_cfg_get_stop_on_failure_attempts();
```

Comment:
For the time being, only the platform manager thread should make bpl calls. That should be changed because we want e.g. controller to also be able to access bpl - cfr. #542. 

As it happens, though, the platform manager doesn't run in a separate thread (even though it's called main_thread, the thread is never start()-ed). Therefore, we *can* access bpl from this function. Only, we have to make sure it's done after bpl_init(). Therefore, the entire code flow below should be changed:

- platform_mgr.init()
- read stop_on_failure_attempts
- construct and start backhaul_mgr
- run the `g_running` loop

Reply:
Done